cmake/f3dVersion.cmake

```diff
@@ -0,0 +1,86 @@
+# Modified from VTKDetermineVersion.cmake from VTK project
+# Copyright (c) Kitware, Inc.
+# License BSD 3 Clause
+
+# Used to determine the version for F3D source using "git describe", if git
+# is found. On success sets following variables in caller's scope:
+#   ${var_prefix}_VERSION
+#   ${var_prefix}_MAJOR_VERSION
+#   ${var_prefix}_MINOR_VERSION
+#   ${var_prefix}_PATCH_VERSION
+#   ${var_prefix}_PATCH_VERSION_EXTRA
+#   ${var_prefix}_VERSION_FULL
+#   ${var_prefix}_VERSION_IS_RELEASE is true, if patch-extra is empty.
+#
+# If git is not found, or git describe cannot be run successfully, then these
+# variables are left unchanged and status message is printed.
+#
+# This expect ${var_prefix}_MAJOR_VERSION, ${var_prefix}_MINOR_VERSION, ${var_prefix}_PATCH_VERSION
+# and ${var_prefix}_VERSION to be already set and will be used for checking when possible.
+#
+# Arguments are:
+#   source_dir : Source directory
+#   git_command : git executable
+#   var_prefix : prefix for variables e.g. "f3d".
+function(determine_version source_dir git_command var_prefix)
```

Comment:
Like `f3d_embed_file`, should we prefix the CMake functions in this file by `f3d_`?